url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/entities/people/Activity.java

```diff
@@ -0,0 +1,841 @@
+		if(appliances.get(day) == null) {
+			appliances.put(day, new Vector<Appliance>());
+		}
+		Vector<Appliance> vector = appliances.get(day);
+		vector.add(a);
+		if(probApplianceUsed.get(day) == null) {
+			probApplianceUsed.put(day, new Vector<Double>());
+		}
+		Vector<Double> probVector = probApplianceUsed.get(day);
+		probVector.add(prob);
+	}
+
+	/**
+	 * Adds a start time probability distribution to the specified activity model.
+	 *
+	 * @param day the activity model
+	 * @param probDist the probability distribution
+	 */
+	public void addStartTime(String day, ProbabilityDistribution probDist) {
+		probStartTime.put(day, probDist);
+	}
+
+	/**
+	 * Adds a duration probability distribution to the specified activity model.
+	 *
+	 * @param day the activity model
+	 * @param probDist the probability distribution
+	 */
+	public void addDuration(String day, ProbabilityDistribution probDist) {
+		probDuration.put(day, probDist);
+	}
+
+	/**
+	 * Adds a repetitions probability distribution to the specified activity model.
+	 *
+	 * @param day the activity model
+	 * @param probDist the probability distribution
+	 */
+	public void addTimes(String day, ProbabilityDistribution probDist) {
+		nTimesGivenDay.put(day, probDist);
+	}
+
+	/**
+	 * Adds the shiftable property for the specified activity model.
+	 *
+	 * @param day the activity model
+	 * @param value the value for the shiftable property
+	 */
+	public void addShiftable(String day, Boolean value) {
+		shiftable.put(day, new Boolean(value));
+	}
+
+	/**
+	 * Adds the config property for the specified activity model.
+	 *
+	 * @param day the activity model
+	 * @param value the value for the config property
+	 */
+	public void addConfig(String day, Boolean value) {
+		config.put(day, new Boolean(value));
+	}
+
+	/* (non-Javadoc)
+	 * @see eu.cassandra.sim.entities.Entity#getName()
+	 */
+	@Override
+	public String getName () {
+		return name;
+	}
+
+	/* (non-Javadoc)
+	 * @see eu.cassandra.sim.entities.Entity#getDescription()
+	 */
+	@Override
+	public String getDescription () {
+		return description;
+	}
+
+	/* (non-Javadoc)
+	 * @see eu.cassandra.sim.entities.Entity#getType()
+	 */
+	@Override
+	public String getType () {
+		return type;
+	}
+
+	/**
+	 * Gets the simulation world.
+	 *
+	 * @return the simulation world
+	 */
+	public SimulationParams getSimulationWorld () {
+		return simulationWorld;
+	}
+
+	/**
+	 * Gets the key.
+	 *
+	 * @param keyword the keyword
+	 * @return the key
+	 */
+	private String getKey(String keyword) {
+		Set<String> set = nTimesGivenDay.keySet();
+		Iterator<String> iter = set.iterator();
+		while(iter.hasNext()) {
+			String key = iter.next();
+			if(key.contains(keyword)) {
+				return key;
+			}
+		}
+		return new String();
+	}
+
+	/**
+	 * Calculate expected power per minute for a full day.
+	 *
+	 * @return the expected power values per minute for a full day
+	 */
+	public double[] calcExpPower() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 437-457 into a separate method.